src/distilabel/llm/google/vertexai.py

```diff
@@ -0,0 +1,517 @@
+        num_threads: Union[int, None] = None,
+    ) -> None:
+        """Initializes the `VertexGenerativeModelLLM` class.
+
+        Args:
+            task (Task): the task to be performed by the LLM.
+            model (str, optional): the model to be used for generation. Defaults to "gemini-pro".
+            temperature (float, optional): the temperature to be used for generation.
+                Defaults to 1.0.
+            top_p (float, optional): the top-p value to be used for generation.
+                Defaults to 1.0.
+            top_k (int, optional): the top-k value to be used for generation.
+                Defaults to 40.
+            max_new_tokens (int, optional): the maximum number of tokens to be generated.
+                Defaults to 128.
+            num_threads (Union[int, None], optional): the number of threads to be used
+                for parallel generation. If `None`, no parallel generation will be performed.
+                Defaults to `None`.
+        """
+        super().__init__(task=task, num_threads=num_threads)
+
+        if not _VERTEXAI_AVAILABLE:
+            raise ImportError(
+                "`VertexAILLM` cannot be used as `google-cloud-aiplatform` is not installed,"
+                " please install it with `pip install google-cloud-aiplatform`"
+            )
+
+        self.temperature = temperature
+        self.top_p = top_p
+        self.top_k = top_k
+        self.max_output_tokens = max_new_tokens
+        self.stop_sequences = stop_sequences
+
+        if is_gemini_model(model):
+            self.model = GenerativeModel(model)
+        elif is_codey_model(model):
+            self.model = CodeGenerationModel.from_pretrained(model)
+        else:
+            self.model = TextGenerationModel.from_pretrained(model)
+
+    @property
+    def model_name(self) -> str:
+        """Returns the name of the model used for generation."""
+        if isinstance(self.model, GenerativeModel):
+            return self.model._model_name
+
+        return self.model._model_id
+
+    def _generate_contents(
+        self, inputs: List[Dict[str, Any]]
+    ) -> List[List[Dict[str, Any]]]:
+        """Generates a list of valid dicts that can be parsed to `vertexai.preview.generative_models.Content`
+        objects for each input.
+
+        Args:
+            inputs (List[Dict[str, Any]]): the inputs to be used for generation.
+
+        Returns:
+            List[List[Dict[str, Any]]]: the list of valid `vertexai.preview.generative_models.Content`
+                objects.
+        """
+        contents = []
+        for input in inputs:
+            prompt = self.task.generate_prompt(**input)
+            contents.append(
+                [
+                    {"role": "user", "parts": [{"text": prompt.system_prompt}]},
+                    {"role": "model", "parts": [{"text": "Understood."}]},
+                    {"role": "user", "parts": [{"text": prompt.formatted_prompt}]},
```

Comment:
I'm a bit confused here, is the `model` message really required? Isn't it better to just concatenate the `system_prompt` and the `formatted_prompt` using a line-break instead as the only `user` message?

Reply:
Yes, that works too.